js/src/volume.js

```diff
@@ -279,6 +282,20 @@ var ScatterView = widgets.WidgetView.extend( {
         _.mapObject(this.model.changedAttributes(), function(val, key){
             console.log("changed " +key)
             this.previous_values[key] = this.model.previous(key)
+            if (key == "index"){
+              pindex = this.previous_values["index"]
+              this.previous_values["x"] = this.model.previous("x")[pindex]
+              this.previous_values["y"] = this.model.previous("y")[pindex]
+              this.previous_values["z"] = this.model.previous("z")[pindex]
```

Comment:
Ok, but I think we should better do
```python
if(_.isArray(this.model.previous('x')) { // changing index effectively changes the 'x' that the shader sees
  this.previous_values["x"] = this.model.previous('x')[this.model.previous('sequence_index')]
  this.attributes_changed['x'] = ['x']
} //
because the previous index was indexing the previous x 
```